File: CS/DevExpressMvcApplication1/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpress.XtraScheduler.Xml;
using DevExpressMvcApplication1.Models;
using DevExpressMvcApplication1.Helpers;

namespace DevExpressMvcApplication1.Controllers
{
    public class HomeController : Controller
    {
        //
        // GET: /Home/

        public ActionResult Index()
        {
            return View(SchedulerDataHelper.DataObject);
        }

        public ActionResult SchedulerPartial() {
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);        
        }

        public ActionResult EditAppointment() {
            UpdateAppointment();
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }

        public ActionResult ContactPartial(int CompanyID) {
            ViewBag.ContactsDataSource = SchedulerDataHelper.GetCompanyContacts(CompanyID);
            return PartialView("ComboboxContactPartial");
        }
        public ActionResult CompanyPartial() {
            return PartialView("ComboboxCompanyPartial");
        }

        static void UpdateAppointment() {
            List<CustomAppointment> appointmnets = System.Web.HttpContext.Current.Session["AppointmentsList"] as List<CustomAppointment>;
            List<CustomResource> resources = System.Web.HttpContext.Current.Session["ResourcesList"] as List<CustomResource>;

            CustomAppointment[] insertedAppts = SchedulerExtension.GetAppointmentsToInsert<CustomAppointment>(SchedulerDataHelper.GetSchedulerSettings(), appointmnets, resources);
            SchedulerDataHelper.InsertAppointments(insertedAppts);


            CustomAppointment[] updatedAppts = SchedulerExtension.GetAppointmentsToUpdate<CustomAppointment>(SchedulerDataHelper.GetSchedulerSettings(), appointmnets, resources);
            SchedulerDataHelper.UpdateAppointments(updatedAppts);

            CustomAppointment[] removedAppts = SchedulerExtension.GetAppointmentsToRemove<CustomAppointment>(SchedulerDataHelper.GetSchedulerSettings(), appointmnets, resources);
            SchedulerDataHelper.RemoveAppointments(removedAppts);
        }


    }
}

File: VB/DevExpressMvcApplication1/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpressMvcApplication1.Models
Imports DevExpressMvcApplication1.Helpers

Namespace DevExpressMvcApplication1.Controllers

    Public Class HomeController
        Inherits Controller

        '
        ' GET: /Home/
        Public Function Index() As ActionResult
            Return View(SchedulerDataHelper.DataObject)
        End Function

        Public Function SchedulerPartial() As ActionResult
            Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
        End Function

        Public Function EditAppointment() As ActionResult
            Call UpdateAppointment()
            Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
        End Function

        Public Function ContactPartial(ByVal CompanyID As Integer) As ActionResult
            ViewBag.ContactsDataSource = SchedulerDataHelper.GetCompanyContacts(CompanyID)
            Return PartialView("ComboboxContactPartial")
        End Function

        Public Function CompanyPartial() As ActionResult
            Return PartialView("ComboboxCompanyPartial")
        End Function

        Private Shared Sub UpdateAppointment()
            Dim appointmnets As List(Of CustomAppointment) = TryCast(Web.HttpContext.Current.Session("AppointmentsList"), List(Of CustomAppointment))
            Dim resources As List(Of CustomResource) = TryCast(Web.HttpContext.Current.Session("ResourcesList"), List(Of CustomResource))
            Dim insertedAppts As CustomAppointment() = SchedulerExtension.GetAppointmentsToInsert(Of CustomAppointment)(SchedulerDataHelper.GetSchedulerSettings(), appointmnets, resources)
            Call SchedulerDataHelper.InsertAppointments(insertedAppts)
            Dim updatedAppts As CustomAppointment() = SchedulerExtension.GetAppointmentsToUpdate(Of CustomAppointment)(SchedulerDataHelper.GetSchedulerSettings(), appointmnets, resources)
            Call SchedulerDataHelper.UpdateAppointments(updatedAppts)
            Dim removedAppts As CustomAppointment() = SchedulerExtension.GetAppointmentsToRemove(Of CustomAppointment)(SchedulerDataHelper.GetSchedulerSettings(), appointmnets, resources)
            Call SchedulerDataHelper.RemoveAppointments(removedAppts)
        End Sub
    End Class
End Namespace
